.github/workflows/license.yml

```diff
@@ -0,0 +1,70 @@
+name: License Check
+
+on: pull_request
+
+jobs:
+  license_job:
+    runs-on: ubuntu-latest
+    name: Run license checks on patch series (PR)
+    steps:
+    - name: Checkout the code
+      uses: actions/checkout@v2
+      with:
+        path: ncs/nrf
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 0
+
+    - name: cache-pip
+      uses: actions/cache@v1
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-doc-pip
+
+    - name: Install python dependencies
+      working-directory: ncs/nrf
+      run: |
+        pip3 install -U pip
+        pip3 install -U -r scripts/ci/requirements.txt
+        export PATH="$HOME/.local/bin:$PATH"
+        grep -E "west" scripts/requirements-fixed.txt | xargs pip3 install -U
+        pip3 show -f west
+
+    - name: West init and update
+      env:
+        BASE_REF: ${{ github.base_ref }}
+      working-directory: ncs/nrf
+      run: |
+        export PATH="$HOME/.local/bin:$PATH"
+        export PATH="$HOME/bin:$PATH"
+        git config --global user.email "you@example.com"
+        git config --global user.name "Your Name"
+        git remote -v
+        git rebase origin/${BASE_REF}
+        # debug
+        git log  --pretty=oneline | head -n 10
+        west init -l .
+        west update
```

Comment:
Should it be run also on normally hide modules as find-my and homekit?

Reply:
we should as @kapi-no and @maciejpietras if they want this on their projects.

Reply:
@grochu says they want it on Find-my.

Reply:
generally yes, with remark that by doing it we do not accidentally share information about content (even files names etc) to external users

Reply:
And we also would like to have it for another private repo https://github.com/nrfconnect/sdk-sidewalk with the same remark as for homekit (so not sharing any data outside)

Reply:
If I understand correctly if we add find-my and homekit here it could be visible for every PR. Then maybe license check for closed repos should be done on PR level but somehow inside this repos.

Reply:
Speaking for find-my, I think we should run the check. This should be consistent through the whole NCS tree.

Reply:
@thst-nordic  @carlescufi do you know if github actions and reports from it are available for external contributors? If yes we can not use sbom for private repos here.

Reply:
They are. I can see action logs and results from private account outside the organization.

Reply:
Ok, then we can't add any privat repo here.

Reply:
I only found such approach to hide secret information https://www.tutorialworks.com/github-actions-mask-url/
basically you can have a gh secret storing strings one likes to hide and then whenever they are detected in the logs they are masked out. I think it won't be useful in our case since, if I got it correctly, it requires in-advance knowledge of what to hide

Reply:
This workflow is not applied to private repos and I've never intended to do so. If private repo is modified, we see only hash change in manifest file, so this workflow will not check any license.

The simplest solution that I see is to add almost identical workflow to each repository where we want to check licenses including private repos. In this case, no information leak can happen.

I've already prepared such workflow for nrfxlib and find-my and I am planning to create PRs for them as soon as this PR goes in.